src/mahoji/lib/abstracted_commands/nightmareZoneCommand.ts

```diff
@@ -0,0 +1,482 @@
+import { ChatInputCommandInteraction } from 'discord.js';
+import { calcWhatPercent, reduceNumByPercent, round, sumArr, Time } from 'e';
+import { Bank } from 'oldschooljs';
+
+import { MAX_QP } from '../../../lib/constants';
+import { trackLoot } from '../../../lib/lootTrack';
+import { resolveAttackStyles } from '../../../lib/minions/functions';
+import { getMinigameEntity } from '../../../lib/settings/minigames';
+import { SkillsEnum } from '../../../lib/skilling/types';
+import { Skills } from '../../../lib/types';
+import { formatDuration, hasSkillReqs, stringMatches } from '../../../lib/util';
+import addSubTaskToActivityTask from '../../../lib/util/addSubTaskToActivityTask';
+import { calcMaxTripLength } from '../../../lib/util/calcMaxTripLength';
+import getOSItem from '../../../lib/util/getOSItem';
+import { updateBankSetting } from '../../../lib/util/updateBankSetting';
+import { handleMahojiConfirmation } from '../../mahojiSettings';
+import { NightmareZoneActivityTaskOptions } from './../../../lib/types/minions';
+
+export const NMZ_STRATEGY = ['experience', 'points'] as const;
+export type NMZStrategy = typeof NMZ_STRATEGY[number];
```

Comment:
It'd be a good idea to move this to `constants.ts`, because it usually creates problems to import things from command files.